apiserver/facades/client/spaces/remove.go

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2020 Canonical Ltd.
+ * Licensed under the AGPLv3, see LICENCE file for details.
+ */
+
+package spaces
+
+import (
+	"github.com/juju/errors"
+	"gopkg.in/juju/names.v3"
+	"gopkg.in/mgo.v2/txn"
+
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/controller"
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/core/permission"
+)
+
+// RemoveSpace describes a space that can be removed.
+type RemoveSpace interface {
+	Refresh() error
+	RemoveSpaceOps() []txn.Op
+}
+
+type spaceRemoveModelOp struct {
+	space   RemoveSpace
+	subnets []Subnet
+}
+
+type Subnet interface {
+	MoveSubnetOps(spaceID string) []txn.Op
+}
+
+func (o *spaceRemoveModelOp) Done(err error) error {
+	return err
+}
+
+func NewRemoveSpaceModelOp(space RemoveSpace, subnets []Subnet) *spaceRemoveModelOp {
+	return &spaceRemoveModelOp{
+		space:   space,
+		subnets: subnets,
+	}
+}
+
+func (sp *spaceRemoveModelOp) Build(attempt int) ([]txn.Op, error) {
+	var totalOps []txn.Op
+
```

Comment:
not sure if the below should be added in this patch:
in a later patch we would need to ensure the following (?):
- transaction ordering -> subnets added to space, before space get;s deleted (refcounting,alive/dead..)
- update the spaceID on that subnet struct (`subet.spaceID = alphaID`)
- maybe use ModelOperation `Done(error) error`  and do some of the above, like update spaceID and ensure other things?

Reply:
OK to leave these for now, I think.